plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1415,9 +1429,28 @@ private void finishWrite(ConnectorSession session, IcebergTableHandle table, Col
             cleanExtraOutputFiles(session, writtenFiles.build());
         }
 
+        if (!fullyDeletedFiles.isEmpty()) {
+            try {
+                FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), new Path(table.getTableLocation()));
+                for (List<CommitTaskData> commitTasksToCleanUp : fullyDeletedFiles.values()) {
+                    for (CommitTaskData commitTaskData : commitTasksToCleanUp) {
+                        fileSystem.delete(new Path(commitTaskData.getPath()), false);
+                    }
+                }
+            }
+            catch (IOException e) {
+                log.warn(e, "Failed to clean up uncommitted position delete files");
```

Comment:
probably because we create delete files for this few lines below? 
Tough I'd expect propagation too.

Reply:
Failing to delete a file that is not going to be committed didn't seem like enough of a problem to warrant failing the query.

If we have limited fs permissions and can't delete files, for example, we would still be able to write deletes.

This would eventually get picked up by a `remove_orphan_files` collection